cmd/collector/app/http_handler.go

```diff
@@ -42,19 +38,20 @@ var (
 // APIHandler handles all HTTP calls to the collector
 type APIHandler struct {
 	jaegerBatchesHandler JaegerBatchesHandler
+	cfgHandler           clientcfghttp.HTTPHandler
 }
 
 // NewAPIHandler returns a new APIHandler
-func NewAPIHandler(
-	jaegerBatchesHandler JaegerBatchesHandler,
-) *APIHandler {
+func NewAPIHandler(jaegerBatchesHandler JaegerBatchesHandler, cfgHandler clientcfghttp.HTTPHandler) *APIHandler {
 	return &APIHandler{
 		jaegerBatchesHandler: jaegerBatchesHandler,
+		cfgHandler:           cfgHandler,
 	}
 }
 
 // RegisterRoutes registers routes for this handler on the given router
 func (aH *APIHandler) RegisterRoutes(router *mux.Router) {
+	aH.cfgHandler.RegisterRoutes(router)
```

Comment:
I think it would be better to mount the endpoints under`/api` prefix, to match `/api/traces`. To do that we need to extend cfgHandler struct to accept a `BasePath` parameter via `HTTPHandlerParams`.